src/lib/alloc.c

```diff
@@ -437,7 +437,7 @@ static void *get_ptr_from_heap(struct mm_heap *heap, int zone, uint32_t caps,
 static void free_block(void *ptr)
 {
 	struct mm_heap *heap;
-	struct block_map *block_map;
+	struct block_map *block_map = NULL;
```

Comment:
This should be a seperate PR as it's good to merge now and a bug fix.

Reply:
@lgirdwood This can only be triggered with -Os build and I do not know why.
I can create a another PR for the bug fixes and keep XCC enable in this.